lib/shopify_api/resources/product.rb

```diff
@@ -16,6 +16,15 @@ def price_range
       end
     end
 
+    def total_inventory=(new_value)
+      raise_deprecated_inventory_call('total_inventory') unless allow_inventory_params?
+      super
+    end
+
+    def serializable_hash(options = {})
+      allow_inventory_params? ? super(options) : super(options).except('total_inventory')
```

Comment:
Changing to
`allow_inventory_params? ? super(options) : super(options).tap{|s| s['variant'].except!('inventory_quantity', 'old_inventory_quantity')}`
Fixes this commit.